Remove Base64EncodePolicy

annevk opened the pull request:
#### 76cc35217b8618b6c07fd31e05e07c0e6ea61339
<pre>
Remove Base64EncodePolicy
<a href="https://bugs.webkit.org/show_bug.cgi?id=255412">https://bugs.webkit.org/show_bug.cgi?id=255412</a>
rdar://108010162

Reviewed by Myles C. Maxfield and Alex Christensen.

Everything Base64EncodePolicy is used for can be done by Base64EncodeMap.

Also make some nearby enums use bool as underlying type.

* Source/WTF/wtf/text/Base64.cpp:
(WTF::base64EncodeInternal):
* Source/WTF/wtf/text/Base64.h:
(WTF::calculateBase64EncodedSize):

Canonical link: <a href="https://commits.webkit.org/262987@main">https://commits.webkit.org/262987@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/09c5a32a158eb0db100316092aefb84df54b72a2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/3145 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/3205 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/3315 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/4557 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/3503 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/3266 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/3232 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/2735 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/3176 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/3492 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/2822 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/4363 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/990 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/2797 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/2638 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/2591 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/2768 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/2848 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/4105 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/2968 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/3220 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/2568 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/3212 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/2805 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/2799 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/784 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/790 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/2797 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/3294 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/3064 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/906 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

litherum commented on Source/WTF/wtf/text/Base64.h:
```diff
@@ -34,30 +34,27 @@
 };
 
-enum class Base64DecodeMap { Default, URL };
+enum class Base64DecodeMap : bool { Default, URL };
```
is `bool` better than `uint8_t`? IIRC best practice was to use uint8_t because of some quirk of MSVC

annevk replied:
I modeled this after some internal discussion and commits such as cebd0f9727d7.

litherum reviewed (approved):
r=me assuming it is, actually, unused

annevk committed:
Remove Base64EncodePolicy
https://bugs.webkit.org/show_bug.cgi?id=255412
rdar://108010162

Reviewed by Myles C. Maxfield and Alex Christensen.

Everything Base64EncodePolicy is used for can be done by Base64EncodeMap.

Also make some nearby enums use bool as underlying type.

* Source/WTF/wtf/text/Base64.cpp:
(WTF::base64EncodeInternal):
* Source/WTF/wtf/text/Base64.h:
(WTF::calculateBase64EncodedSize):

Canonical link: https://commits.webkit.org/262987@main
